### ISSUE TITLE
Reproducing the plot with confidence intervals with multivariate data

### ISSUE THREAD

**Issue by hiyamgh** [stanfordmlgroup/ngboost#138]
First, Thank you for this very awesome repository and the work done is indeed fantastic.

I want to assess the effectiveness of my probabilistic predictions with ngboost but my **data is multivariate**, more specifically, I have 19 columns as input variables and 1 column as the target variable. I want to produce a plot like follows:
![Screenshot (651)](https://user-images.githubusercontent.com/35295550/85771924-0b54c800-b725-11ea-8384-3bfd2aa793dd.png)

I am referring to the [toy_single.py code](https://github.com/stanfordmlgroup/ngboost/blob/master/figures/toy_single.py) which clearly produces this plot very well. I have a few questions:

 
1.  Why do the yellow dots represent the **training** ? I assume we are doing the predictions on the testing data and thus these points must represent rather the testing **point** predictions (I think I might be mistaken here). Please clarify.
2. The yellow points (related to bullet number 1) are produced using this line: ```plt.scatter(x_tr[:, 1], y_tr, color="black", marker=".", alpha=0.5)```. With what should I replace ```x_tr[:, 1]``` if I have multivariate data ?
3.  Here as well: 
``` 
plt.plot(
        x_te[:, 1],
        preds.loc,
        color="black",
        linestyle="-",
        linewidth=1,
        label="Predicted mean",
    )
plt.plot(
        x_te[:, 1],
        preds.loc - 1.96 * preds.scale,
        color="black",
        linestyle="--",
        linewidth=0.3,
        label="95\% prediction interval",
    )
plt.plot(
        x_te[:, 1],
        preds.loc + 1.96 * preds.scale,
        color="black",
        linestyle="--",
        linewidth=0.3,
    )
```
With what should I replace ```x_te[:, 1]``` because, again, my data is multivariate 

4. Should I always use this ```calibration_regression(preds, y_test)```  function for producing the plot above ?

My code so far 
```
import pandas as pd
from sklearn.preprocessing import MinMaxScaler
from ngboost.evaluation import *
from argparse import ArgumentParser
from ngboost.distns import Normal
from ngboost import NGBoost
from ngboost.scores import MLE
from ngboost.learners import default_tree_learner

# read the training + testing data
df_train = pd.read_csv('../input/all_without_date/collated/all_columns/df_train_collated.csv')
df_test = pd.read_csv('../input/all_without_date/collated/all_columns/df_test_collated.csv')
target_variable = 'demand'

X_train = np.array(df_train.loc[:, df_train.columns != target_variable])
y_train = np.array(df_train.loc[:, target_variable])

X_test = np.array(df_test.loc[:, df_test.columns != target_variable])
y_test = np.array(df_test.loc[:, target_variable])

# min max scaling
scaler = MinMaxScaler()
X_train = scaler.fit_transform(X_train)
X_test = scaler.transform(X_test)

argparser = ArgumentParser()
argparser.add_argument("--n-estimators", type=int, default=301)
argparser.add_argument("--lr", type=float, default=0.03)
argparser.add_argument("--minibatch-frac", type=float, default=0.1)
argparser.add_argument("--natural", action="store_true")
args = argparser.parse_args()

ngb = NGBoost(
    Base=default_tree_learner,
    Dist=Normal,
    Score=MLE,
    n_estimators=args.n_estimators,
    learning_rate=args.lr,
    natural_gradient=args.natural,
    minibatch_frac=args.minibatch_frac,
    verbose=True,
)

ngb.fit(X_train, y_train)
```

Much appreciated!

**Comment by tonyduan** [comment]
@hiyamgh Our toy example was on univariate data (one feature), so it's straightforward to plot points (x, y) onto a 2D plane. In general (to my knowledge) it's not obvious how to visualize multi-dimensional regression when x has dimension greater than 2. Something you could try is PCA or t-SNE on the multi-dimensional features to reduce dimensionality of x before visualization to 2D, then make a 3D plot including y. 

1. For the purpose of assessing prediction on the test set, you're right that the plot should be of the test set. But our purpose in making that visualization is slightly different. Our purpose is to show that we can accurately capture the heteroskedasticity in the prediction intervals. Since the training set is distributed identically to the test set, the ground truth heteroskedasticity is similar across the two splits. Probably the more principled way to do this would be to simulate a larger test set and plot that instead, but either way the visualization captures the idea we wanted to get across.

2. See my earlier comments.

3. See my earlier comments.

4. Actually I think we don't even use the results of calling the `calibration_regression` function in our toy code, so you can ignore it for the visualization.

**Comment by alejandroschuler** [comment]
@hiyamgh also wanted to note that the bands that are shown in the plot are [prediction intervals, not confidence intervals](https://towardsdatascience.com/interpreting-the-probabilistic-predictions-from-ngboost-868d6f3770b2).
